themes/bootstrap3/js/advanced_search.js

```diff
@@ -1,4 +1,5 @@
 /*exported addGroup, addSearch, deleteGroup, deleteSearch */
+/* jshint latedef: nofunc */
```

Comment:
This is the only thing that gives me pause. Is it necessary to change jshint settings for just this file?

Reply:
I do think so... because deleteGroup calls addGroup and addGroup have deleteGroup in one of click handlers, which makes jshint complain.... I would like to not add this, but I was not able to find out other solution...

Reply:
I think commit 4a388c0ab83c9e20e33c33beec233cec11855d90 avoids the error in a way more consistent with some of VuFind's other JS code while maintaining the same functionality. I'll let @crhallberg take another look at it before I merge, but I think it's okay now.

It would be better to refactor this code as a self-contained VuFind module rather than loose functions, but that's a fairly disruptive change to make at this stage in the game, so I think I can live with the current modifications for now, and we can revisit the rest for VuFind 8!

Reply:
Thanks @demiankatz!